js/DataService.js

```diff
@@ -288,16 +287,25 @@ angular.module('PathOfDamage')
 
   function decodeElements(encodedElements) {
     if (encodedElements) {
-      var elements = [];
-      var booleans = parseInt(encodedElements, 36).toString(2).split('');
-      var index = booleans.length;
-      for (var i = ELEMENTS.length; i-- > 0;) {
-        elements[ELEMENTS[i]] = decodeBoolean(booleans[--index]);
-      }
+      var booleans = decodeBooleans(encodedElements, ELEMENTS.length);
+      var elements = {};
+      ELEMENTS.forEach(function (element, index) {
+        elements[element] = booleans[index];
+      });
       return elements;
     }
   }
 
+  function decodeBooleans(encoded, length) {
+    var booleans = [];
+    var booleanString = parseInt(encoded, 36).toString(2);
+    var stringLength = booleanString.length;
+    for (var i = 0; i < length; i++) {
+      booleans[i] = decodeBoolean(booleanString[--stringLength]);
+    }
+    return booleans;
```

Comment:
I also couldn't see any more concise way to do this silly logic with all the extra initialization involved.